### ISSUE TITLE
false negative for SC2034 when testing associative array key existence with a variable

### ISSUE THREAD

**Issue by jmcantrell** [koalaman/shellcheck#2309]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

declare -A array

key=a

if [[ -v array[$key] ]]; then
        echo yes
else
        echo no
fi
```

#### Here's what shellcheck currently says:
```
Line 5	SC2034: key appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

Nothing.

**Comment by koalaman** [comment]
No no, it was an active bug when the issue was posted, and I fixed and closed it :+1:

**Comment by jmcantrell** [comment]
Oh ok. Thank you!
